core/src/main/scala/org/apache/spark/sql/execution/ConnectionPool.scala

```diff
@@ -132,8 +134,20 @@ object ConnectionPool {
    */
   def getPoolConnection(id: String, dialect: JdbcDialect,
       poolProps: Map[String, String], connProps: Properties,
-      hikariCP: Boolean): Connection = {
-    val ds = getPoolDataSource(id, poolProps, connProps, hikariCP)
+      hikariCP: Boolean, urlSuffix: String): Connection = {
+    val url = poolProps.get("url")
+    val poolPropsSecure = if (url.isDefined) {
+      val urlSecure = if (id.startsWith(Constant.SHADOW_SCHEMA_NAME)) {
+        val bootProperties = Misc.getMemStore.getBootProperties
+        if (bootProperties.containsKey("user") && bootProperties.containsKey("password")) {
+          url.get + ";user=" + bootProperties.get("user") +
+              ";password=" + bootProperties.get("password") +
+              ";default-schema=" + Misc.SNAPPY_HIVE_METASTORE + ";"
```

Comment:
Why setting DBA properties here?? This won't work for smart connector. The user/password should be coming as part of connProps.